tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import ssl
+import warnings
+from typing import Any, Tuple, cast
+
+import pytest
+
+from aiohttp import TCPConnector, web
+from aiohttp.client import ServerDisconnectedError
+from aiohttp.pytest_plugin import AiohttpClient, AiohttpServer
+from aiohttp.test_utils import TestClient, TestServer
+
+
+async def handle_root(request: web.Request) -> web.Response:
+    """respond after server was restarted, forcing connection_lost"""
+    cq = request.app["cq"]
+    dq = request.app["dq"]
+
+    await dq.put(0)
+
+    await cq.get()
+    cq.task_done()
+
+    return web.Response(text="")
+
+
+async def _prepare(
+    aiohttp_server: Any,
+    aiohttp_client: Any,
+    ssl_ctx: ssl.SSLContext,
+    client_ssl_ctx: ssl.SSLContext,
+    cq: "asyncio.Queue[int]",
+    dq: "asyncio.Queue[int]",
+) -> Tuple[TestServer, TestClient]:
+    app = web.Application()
+    app["cq"] = cq
+    app["dq"] = dq
+    app.router.add_get("/", handle_root)
+    server = await aiohttp_server(app, port=0, ssl=ssl_ctx)
+
+    session = await aiohttp_client(server, connector=TCPConnector(ssl=client_ssl_ctx))
+
+    # replace SockSite … it is different and blocks
+    await dq.put(0)
+    await _restart(server.runner, ssl_ctx, session.port, cq, dq)
+    await cq.get()
+
+    return server, session
+
+
+async def _close(server: TestServer, session: TestClient) -> None:
+    await server.close()
+    await session.close()
+    await server
```

Comment:
I don't think that fixes the issue, but somehow masks it.
```
TypeError: object TestServer can't be used in 'await' expression
```

Reply:
I'm also seeing:
```
Exception ignored in: <coroutine object aiohttp_client.<locals>.finalize at 0x7fe61a202340>
Traceback (most recent call last):
  File "<string>", line 1, in <lambda>
KeyError: '__import__'
Exception ignored in: <coroutine object aiohttp_client.<locals>.finalize at 0x7fe61a202340>
Traceback (most recent call last):
  File "<string>", line 1, in <lambda>
KeyError: '__import__'
```
Which must be related to the problem.

Reply:
Think I've got a solution. The server/client already auto-close in the fixture, but the test was mixing up loops.

Reply:
So, actually, the test passes on 3.11+, which is why the warning didn't happen on the other tests.